src/FileMigrationHelper.php

```diff
@@ -143,12 +155,31 @@ protected function migrateFile($base, File $file, $legacyFilename)
             $file->setFilename($result['Filename']);
         }
 
+        // set original 'LastEdited' date if static is enabled
+        $lastEditedDate = '';
+        if ($this->config()->get('keep_lastedited_date')) {
+            $lastEditedDate = $file->LastEdited;
+        }
+
         // Save and publish
         $file->write();
         if (class_exists(Versioned::class)) {
             $file->copyVersionToStage(Versioned::DRAFT, Versioned::LIVE);
         }
 
+        // if 'LastEdited' date static is enabled, update file's last edited using SQL query
+        if ($this->config()->get('keep_lastedited_date') && $lastEditedDate) {
+            // Draft table
+            $update = SQLUpdate::create('File')->addWhere(['ID' => $file->ID]);
```

Comment:
People can change the name of table in their YML config. So you can't know for a fact that the table will be called `File`.

Use something like this to get the table name. `DataObject::singleton(File::class)->baseTable();`